### ISSUE TITLE
Optaining training errors along training sqeuence?

### ISSUE THREAD

**Issue by Steviey** [tidymodels/tune#287]
**R4.x, Ubuntu 16 LTS**

Hello, how to obtain training errors from tune_grid() along the training process?

Thank you and tidy on.

**Comment by topepo** [comment]
Model errors and warnings are saved in the `. notes` column.

**Comment by Steviey** [comment]
I did not meant regular proceeding errors. I want to plot train errors against test errors. Therefore I need trian errors like from show_best(), but in the order of their occurrence.

**Comment by topepo** [comment]
You can save the predictions using `control_*(save_pred = TRUE)` and, from these, estimate errors/residuals.

**Comment by Steviey** [comment]
Yes, but how to differentiate between testing and training. Everything is under the hood in my example....

```
#Train model on train set and evaluate on test set 
pricing_workflow_fit <- last_fit(pricing_workflow, partObj$splits) 
```

**Comment by topepo** [comment]
I can help more if you can follow the issue template and give us a minimal reproducible example (preferably by using `reprex::reprex()`).

**Comment by Steviey** [comment]
Meanwhile I got an answer from **Andrew Couch**. I'm still fighting tibbles... I kind of hate them (since I saw them the first time).

`It depends on what you mean by train and test errors. 
If you mean the train and test errors within a cross validation or rsample set, 
you would have to write a function to record it as I believe tune_grid and 
fit_resamples only records the final in-sample predictions. 

If you mean train and test errors where train error is the 
final models in-sample error with the train data you can just save 
the collect_metrics() to a data frame and then use last_fit()
and collect_metrics() to extract the test set metrics.

For direct comparisons of predictions between model in-sample and model 
out of sample, you would need to add control_grid(save_pred = True) 
and add it to your fit function. This will save the predictions 
and the actual values in the training data and you could compare 
with the test data predictions.`

**Comment by Steviey** [comment]
After some nights my guess is, last_fit() is not designed to provide in-sample errors. To get them one should use fit_resamples().
